### PR TITLE
deploy: working on real hw remote xPU

### LINKED ISSUE #148: `./scripts/deploy.sh` fails on xPU
```
 $ ./scripts/deploy.sh -m xpu -x 10.246.67.206
Selected mode xpu
Deploying xPU environment
BMC IP address:
Host IP address: 127.0.0.1
xPU IP address: 10.246.67.206
ERROR: Network opi-external declared as external, but could not be found. Please create the network manually using `docker network create opi-external` and try again.
```

### PR DESCRIPTION
**Pull request by glimchb**
Fixes #148

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

File: integration/scripts/deploy.sh [hunk 1]
```diff
@@ -6,9 +6,9 @@
 # environments:
 #
 # * Developer emulated xPU setup
-# * Split setup between an actual xPU and a host/VM/container
+# * Split setup between an actual xPU and a laptop/VM/server
```
Comment (mestery) [inline integration/scripts/deploy.sh:9]: Could be container as well, in fact, I was testing that out locally. But fine for now.

File: integration/scripts/deploy.sh [hunk 2]
```diff
@@ -6,9 +6,9 @@
...
 #
-# This script assumes you have Docker installed on the xPU and host
+# This script assumes you have Docker installed on the xPU and laptop/VM/server
```
Comment (mestery) [inline integration/scripts/deploy.sh:11]: Same container comment here.

File: integration/scripts/deploy.sh [hunk 3]
```diff
@@ -45,17 +45,17 @@ deploy_dev() {
...
-    echo "Host IP address: ${INFRA_IP_ADDRESS}"
+    echo "Infra IP address: ${INFRA_IP_ADDRESS}"
     echo "xPU IP address: ${XPU_IP_ADDRESS}"
 
-    # Deploy to host/VM/container
-    export COMPOSE_FILE=docker-compose.pxe.yml:docker-compose.spdk.yml
-    export DOCKER_HOST="ssh://user@${INFRA_IP_ADDRESS}"
+    # Deploy to laptop/VM/server
+    export COMPOSE_FILE=docker-compose.otel.yml:docker-compose.spdk.yml:docker-compose.pxe.yml:docker-compose.networks.yml
+    [[ -n ${INFRA_IP_ADDRESS} ]] && export DOCKER_HOST="ssh://ubuntu@${INFRA_IP_ADDRESS}"
```
Comment (mestery) [inline integration/scripts/deploy.sh:53]: Fine for now, but we can't hardcode `ubuntu` here.

File: integration/scripts/deploy.sh [hunk 4]
```diff
@@ -45,17 +45,17 @@ deploy_dev() {
...
     bash -c "${DC} up -d"
 
     # Deploy to XPU
-    export COMPOSE_FILE=docker-compose.xpu.yml
-    export DOCKER_HOST="ssh://user@${XPU_IP_ADDRESS}"
+    export COMPOSE_FILE=docker-compose.xpu.yml:docker-compose.networks.yml
+    export DOCKER_HOST="ssh://ubuntu@${XPU_IP_ADDRESS}"
```
Comment (mestery) [inline integration/scripts/deploy.sh:58]: Same comment on `ubuntu` here. We need this to be configurable. Mergable now, but we can fix post-merge.

File: integration/scripts/deploy.sh [hunk 5]
```diff
@@ -65,7 +65,7 @@ MODE=dev
 
 # IP addresses, only used for xpu mode
 XPU_IP_ADDRESS=
-INFRA_IP_ADDRESS=127.0.0.1
+INFRA_IP_ADDRESS=
```
Comment (mestery) [inline integration/scripts/deploy.sh:68]: Why remove the default?

File: integration/scripts/integration.sh [hunk 6]
```diff
@@ -31,6 +31,7 @@ build_containers() {
 start_containers() {
     bash -c "${DC} down"
     docker network prune --force
+    bash -c "${DC} pull"
```
Comment (mestery) [inline integration/scripts/integration.sh:34]: The `up` command should implicitly perform the pull. Does it not?

File: integration/scripts/deploy.sh (hunk 4 above)
Comment (glimchb) [reply integration/scripts/deploy.sh:58]: yep

File: integration/scripts/integration.sh (hunk 6 above)
Comment (glimchb) [reply integration/scripts/integration.sh:34]: nope
if you have something with `latest` tag, up will not re-pull it

**Review by mestery — APPROVED** [review]

**Pull request merged**